app/controllers/passwordless/sessions_controller.rb

```diff
@@ -42,7 +42,7 @@ def show
       BCrypt::Password.create(params[:token])
 
       session = find_session
-      sign_in session.authenticatable
+      sign_in session.authenticatable unless session_expired session
```

Comment:
I don't think we want to continue this method if the session is expired. We could either _1) do a regular old (boring) `if` statement and return early or _2)_ we could create an `Error` class, `raise` that and `rescue` it at the bottom.

Reply:
I have come up with a solution using a custom Error. However, rubocop is saying that the show method is too long/complex... Any ideas how to resolve this? (I started messing around with fixing this by squishing the show method down as much as I could, but I thought I'd ask for your input first, before I commit to refactoring the entire show method)

`Metrics/AbcSize: Assignment Branch Condition size for show is too high. [19.1/15]'
'Metrics/MethodLength: Method has too many lines. [16/10]`

```
def show
  # Make it "slow" on purpose to make brute-force attacks more of a hassle
  BCrypt::Password.create(params[:token])
  session = find_session

  begin
     raise ExpiredSessionError if session.expired?
  rescue ExpiredSessionError
     flash[:error] = 'Your session has expired, please sign in again.'
     redirect_to main_app.root_path && return
  end

  sign_in session.authenticatable

  redirect_enabled = Passwordless.redirect_back_after_sign_in
  destination = reset_passwordless_redirect_location!(User)

  if redirect_enabled && destination
     redirect_to destination
  else
     redirect_to main_app.root_path
  end
end
````

Reply:
For example here is a refactored view: (rubocop is happy with this)

    def show
      # Make it "slow" on purpose to make brute-force attacks more of a hassle
      BCrypt::Password.create(params[:token])

      session = find_session
      raise ExpiredSessionError if session.expired?

      sign_in session.authenticatable

      redirect_enabled = Passwordless.redirect_back_after_sign_in
      target = reset_passwordless_redirect_location!(User)

      redirect_to(redirect_enabled && target ? target : main_app.root_path)
    rescue ExpiredSessionError
      session_expired
    end
    ...
    private
    ...
    def session_expired
      flash[:error] = 'Your session has expired, please sign in again.'
      redirect_to main_app.root_path
    end